linux/datadog_wrapper

```diff
@@ -67,30 +84,78 @@ setUpNodeEnv() {
 setUpDotnetEnv() {
     echo "Setting up Datadog tracing for .NET"
 
-    setUpCommonEnv || return
+    getBinaries || return
 
     if [ -z "$DD_DOTNET_TRACER_VERSION" ]; then
         DD_DOTNET_TRACER_VERSION=2.19.0
     fi
+    
+    echo "Using version $DD_DOTNET_TRACER of the .NET tracer"
     DD_DOTNET_TRACER_FILE=datadog-dotnet-apm-${DD_DOTNET_TRACER_VERSION}.tar.gz
     DD_DOTNET_TRACER_URL=https://github.com/DataDog/dd-trace-dotnet/releases/download/v${DD_DOTNET_TRACER_VERSION}/${DD_DOTNET_TRACER_FILE}
 
     echo "Installing .NET tracer from $DD_DOTNET_TRACER_URL"
-    curl -L $DD_DOTNET_TRACER_URL -o $DD_DOTNET_TRACER_FILE|| return
-    tar -xzf $DD_DOTNET_TRACER_FILE || return
+    curl_or_log_failure $DD_DOTNET_TRACER_URL $DD_DOTNET_TRACER_FILE
 
     export CORECLR_ENABLE_PROFILING=1
     export CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
     export CORECLR_PROFILER_PATH=${DD_DIR}/Datadog.Trace.ClrProfiler.Native.so
     export DD_DOTNET_TRACER_HOME=${DD_DIR}
 }
 
-if [ "$WEBSITE_STACK" = "NODE" ]; then
-    setUpNodeEnv;
-elif [ "$WEBSITE_STACK" = "DOTNETCORE" ]; then
-    setUpDotnetEnv;
-fi
+setUpTraceAgent() {
```

Comment:
Since `setUpTraceAgent` and `setUpDogStatsD` are almost exactly the same, you could combine them into one function that takes the argument `trace-agent` or `dogstatsd`.